### PR TITLE
Gives emitters wires

### PR DESCRIPTION
**Pull request by Time-Green**
# Adds 5 wires to an emitter:

- **Activate**: Activates the emitter. Cut it to prevent activation

- **Deactivate**: Disables the emitter. Cut to prevent deactivation

- **Fire**: Turns the emitter on for 5 seconds, giving 0-3 shots. Cut to disable it's firing mechanism

-_And 2 useless wires_

:cl:
rscadd: Gives emitters wires to: Activate, disable and fire. 
/:cl:

### CODE REVIEW

**Comment by Super3222** [comment]
Interesting. :+1: MERGE

**Comment by ShadowDeath6** [comment]
Cool

**Comment by GrayRachnid** [comment]
butwhy?

**Comment by ShadowDeath6** [comment]
>why

hacking + consistency between machinery

**Comment by Time-Green** [comment]
>Why

So I can set a trap in engi maint to kill all assistants

**Comment by GrayRachnid** [comment]
>hacking + consistency between machinery

If this is about consistency then you've got a huge list of machinery to work on.
Let's forget the rest and focus on engineering based ones only. SMES, PACMANs, Turbine, the tesla thingies, and the other tesla thingies.

>So I can set a trap in engi maint to kill all assistants

What stops you from that now tho
inb4 you don't know that you can already lock emitters so nobody can turn off/on including the AI.

**Comment by ShadowDeath6** [comment]
Power units don't need wires tho, they're all-access

**Comment by Time-Green** [comment]
Gray, it's about utility and fun. Mainly using it in combination with stuff like signallers to create cool combos.
Please give an actual reason instead if repeating "why"

**Comment by matskuman5** [comment]
chrome_2017-05-09_23-32-43

File: code/modules/power/singularity/emitter.dm [hunk 1]
```diff
@@ -168,6 +180,10 @@
 			src.fire_delay = rand(minimum_fire_delay,maximum_fire_delay)
 			src.shot_number = 0
 
+
+		if(!can_fire)
+			visible_message("<span class='danger'>The [src.name] hums!</span>","<span class='userdanger'>You feel a humming.</span>")
```
Comment (ShadowDeath6) [inline code/modules/power/singularity/emitter.dm]: you don't feel humming, can you say hear humming / feel a vibration or something

Comment (Time-Green) [reply code/modules/power/singularity/emitter.dm]: Sure

**Comment by Time-Green** [comment]
I forgot to sync my commits, I'll do that tomorrow

**Comment by Time-Green** [comment]
Synced my commits, tomorrow apparently became 9 days

**Comment by Cruix** [comment]
In council time tomorrow is like 3 months.

**Comment by Cruix** [comment]
BTW I'll merge this tomorrow

**Comment by Cruix** [comment]
You cannot turn off emitters manually, and when you screwdriver the panel closed it runtimes.

**Comment by Time-Green** [comment]
Really? I thought it did...
For futute reference, how do I know something is runtiming? I know what causes it, but not when it does

**Comment by Cruix** [comment]
Look at the dream daemon window (the one with the green circle) and it will print a message like this:

runtime error: 
[09:30:51]bad del
[09:30:51]proc name: qdel (/proc/qdel)
[09:30:51]  source file: garbage.dm,164
[09:30:51]  usr: Adamantine Andesite (/mob/living/carbon/human)
[09:30:51]  src: null
[09:30:51]  usr.loc: the floor (117,158,5) (/turf/open/floor/plasteel/shuttle/purple)
[09:30:51]  call stack:
[09:30:51]qdel(null, 0)
[09:30:51]Emitter (/obj/machinery/power/emitter): attackby(the screwdriver (/obj/item/weapon/screwdriver), Adamantine Andesite (/mob/living/carbon/human), "icon-x=15;icon-y=27;left=1;scr...")
[09:30:51]Adamantine Andesite (/mob/living/carbon/human): ClickOn(Emitter (/obj/machinery/power/emitter), "icon-x=15;icon-y=27;left=1;scr...")
[09:30:51]Emitter (/obj/machinery/power/emitter): Click(the plating (118,159,5) (/turf/open/floor/plating), "mapwindow.map", "icon-x=15;icon-y=27;left=1;scr...")

It might not work if you run your server directly from dream maker. You can open a dream daemon window to run your server by opening the byond client and pressing crtl+D. Remember to set it to trusted and private before you hit run.

**Comment by Time-Green** [comment]
Alrighty, fixed the mess.
Apparently it was going crazy because it had another screwdriver proc: the decon path.
Anyway, runtime and being unable to turn it off is fixed. I also removed the sprites and used the decon ones instead, they look better.

**Pull request merged**